### PR TITLE
User can view specific request

### PR DESCRIPTION
**Pull request by ViaMarcus**
[PT](https://www.pivotaltracker.com/story/show/173332813)

# Feature:
- Users (and visitors) can click on a reQuest to show more information about it, as well as a button for making an offer to help

* Adds a box to view request details.
* Makes request card clickable to load that request into state.
* Shows info about reQuest until a card has been clicked.
* Has logic to render a 'contact' button or message based on if user exists and if user can make an offer on the request (user can't make an offer to a request that he owns or already offered help to)
* Disables contact button if user is not eligible to offer help, and a helpful message to explain why.

# Screenshots:
Screenshot 2020-06-18 at 18 41 23
Screenshot 2020-06-18 at 18 40 30

### CODE REVIEW

**Comment by erikbjoern** [comment]
In the latest PR that's open, Oliver and I added redux state for active request, that can be re-used in this one

**Comment by erikbjoern** [comment]
Also, see the styling convention of my PR, regarding page-container, when that one is merged - basically, the div/component that wraps all the content on any page should have the id="page-container", which puts the content on appropriate distance from header, as well as providing flex-direction: row and default margin between children of page-container.

File: cypress.json [hunk 1]
```diff
@@ -1,3 +1,5 @@
 {
-  "baseUrl": "http://localhost:3001"
+  "baseUrl": "http://localhost:3001",
+  "viewportWidth": 1280,
+  "viewportHeight": 720
```
Comment (erikbjoern) [inline cypress.json:4]: Is this pixels? Seems small, what's the reason?

File: cypress/integration/visitorCanViewSpecificRequest.feature.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  beforeEach(() => {
+    cy.visit('/')
+  });
+
+  describe('as a visitor', () => {
+    beforeEach(() => {
+      cy.stubMain()
+    });
+
+    it('a button should not exist when visitor is browsing', () => {
```
Comment (erikbjoern) [inline cypress/integration/visitorCanViewSpecificRequest.feature.js]: which button?

File: cypress/integration/visitorCanViewSpecificRequest.feature.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      cy.get("#request-1").click()
+      cy.get("#selected-request").within(() => {
+        cy.get("#selected-title").should("contain", "Change to winter tires");
+        cy.get("#selected-description").should("contain", "Need a hand with tyres")
+        cy.get("#selected-requester").should("contain", "someone1@email.com")
+        cy.get("button").should("be.enabled")
+      });
+    });
+  
+    it('a request can have contact button disabled', () => {
```
Comment (erikbjoern) [inline cypress/integration/visitorCanViewSpecificRequest.feature.js]: when is it disabled?

File: cypress.json (hunk 1 above)
Comment (ViaMarcus) [reply cypress.json:4]: Default is 1000x660, so this is bigger, good aspect ratio when styling using cypress testing

**Review by designerofthing — APPROVED** [review]

**Review by erikbjoern — APPROVED** [review]

**Review by kermit-klein — APPROVED** [review]

**Pull request merged**